packages/framer-motion/src/animation/optimized-appear/store.ts

```diff
@@ -4,3 +4,5 @@ export interface AppearStoreEntry {
 }
 
 export const appearAnimationStore = new Map<string, AppearStoreEntry>()
+
+export const elementsWithAppearAnimations = new Set<string>()
```

Comment:
If the appear effect is done, shouldn't we clear the set to avoid memory leaks? (or use `WeakSet/Map` instead?)

Reply:
(The reason they're strings vs Element look ups is partly in case of hydration errors and partly because of pre-hydration checks)

Reply:
ah yeah makes sense I thought we store elements in here, then I guess it's kinda okay-ish